lib/src/revset.rs

```diff
@@ -2123,13 +2151,14 @@ pub struct RevsetCommitIterator<I> {
     iter: I,
 }
 
-impl<I: Iterator<Item = CommitId>> Iterator for RevsetCommitIterator<I> {
+impl<I: RevsetIterator<CommitId>> Iterator for RevsetCommitIterator<I> {
     type Item = BackendResult<Commit>;
```

Comment:
The error type would have to be changed to `RevsetEvaluationError`. If that involves huge code change, please leave a TODO comment.

Reply:
Added a commit for it; it was a relatively small change.